core/emu.py

```
@@ -68,13 +92,46 @@ def watch_emulator(emulator: CoreEmulator) -> int:
+    that the command starts right away, before any debugger has attached.
+
+    Deliberately not `os.execvpe()`: exec skips all of our cleanup, most notably
+    shutting the Tropic01 model down, and a leaked model server silently keeps
+    listening on the same port as the next one.
+    """
+    dbg_process = subprocess.Popen(dbg_command, env=env)
+    # If we are killed before the debugger is, don't leave the emulator behind
+    # holding the Tropic01 model port and the emulator's UDP ports.
+    atexit.register(kill_process, dbg_process)
+    run_process = (
+        subprocess.Popen(run_command, env=env, shell=True) if run_command else None
```

Ruff (0.16.0)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)